public/app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning.tsx

```diff
@@ -1,41 +1,60 @@
 import { css } from '@emotion/css';
+import { isEmpty } from 'lodash';
 import React from 'react';
 
 import { GrafanaTheme2 } from '@grafana/data/src';
 import { Alert, useStyles2 } from '@grafana/ui/src';
 
 import { AlertmanagerChoice } from '../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
 
 interface GrafanaAlertmanagerDeliveryWarningProps {
-  alertmanagerChoice?: AlertmanagerChoice;
   currentAlertmanager: string;
 }
 
-export function GrafanaAlertmanagerDeliveryWarning({
-  alertmanagerChoice,
-  currentAlertmanager,
-}: GrafanaAlertmanagerDeliveryWarningProps) {
+export function GrafanaAlertmanagerDeliveryWarning({ currentAlertmanager }: GrafanaAlertmanagerDeliveryWarningProps) {
   const styles = useStyles2(getStyles);
 
-  if (currentAlertmanager !== GRAFANA_RULES_SOURCE_NAME) {
+  const { useGetAlertmanagerChoiceQuery, useGetExternalAlertmanagersQuery } = alertmanagerApi;
+  const viewingInternalAm = currentAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  const { currentData: alertmanagerChoice } = useGetAlertmanagerChoiceQuery();
+  const { currentData: externalAMs } = useGetExternalAlertmanagersQuery();
+
+  const externalAlertManagers = externalAMs?.activeAlertManagers ?? [];
+
+  if (alertmanagerChoice === AlertmanagerChoice.Internal) {
```

Comment:
Unfortunately due to the rules of hooks I can't think of a way to avoid fetching the external AMs if `Internal` is selected.

Though I _think_ we can do so if we import the deferred version of the `GetExternalAlertmanagersQuery` hook and use a `useEffect` to call it only if we have chosen to send to external AMs. 🤔

Reply:
Well – turns out it's a lot easier to just use `skip` as a config for `useGetExternalAlertmanagersQuery()` 😬